var/spack/repos/builtin/packages/harfbuzz/package.py

```diff
@@ -58,6 +59,20 @@ def flag_handler(self, name, flags):
                 flags.append('-std=gnu99')
         return (None, None, flags)
 
+    def setup_run_environment(self, env):
+        env.prepend_path("GI_TYPELIB_PATH",
+                         join_path(self.prefix.lib, 'girepository-1.0'))
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.prepend_path('XDG_DATA_DIRS', self.prefix.share)
+        env.prepend_path("GI_TYPELIB_PATH",
+                         join_path(self.prefix.lib, 'girepository-1.0'))
+
+    def setup_dependent_run_environment(self, env, dependent_spec):
+        env.prepend_path('XDG_DATA_DIRS', self.prefix.share)
+        env.prepend_path("GI_TYPELIB_PATH",
+                         join_path(self.prefix.lib, 'girepository-1.0'))
+
```

Comment:
I'm not familiar enough with this introspection feature, but is this needed on run time or only on build time? I am asking because it appears to be in `-dev` packages on distributions.

Reply:
Never mind, I see all packages have it in all three places, `run`, `build`, `dependendent_run`.